core/src/test/java/org/frankframework/mongodb/MongoDbSenderTest.java

```diff
@@ -0,0 +1,305 @@
+package org.frankframework.mongodb;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.core.StringContains;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArray;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import lombok.extern.log4j.Log4j2;
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.SenderException;
+import org.frankframework.core.TimeoutException;
+import org.frankframework.mongodb.MongoDbSender.MongoAction;
+import org.frankframework.parameters.Parameter;
+import org.frankframework.senders.SenderTestBase;
+import org.frankframework.stream.Message;
+import org.frankframework.stream.document.DocumentFormat;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Log4j2
+@Testcontainers(disabledWithoutDocker = true)
+@Tag("integration") //Excluded by a default test run, since it requires Docker
+public class MongoDbSenderTest extends SenderTestBase<MongoDbSender> {
```

Comment:
This class I moved from `/test-integration` and improved & fixed it, to get it working. The basis of the unit tests I didn't change, in order to show the previous impl still works.